tests/java/org/pantsbuild/tools/ivy/DependencyUpdateCheckerTest.java

```diff
@@ -0,0 +1,160 @@
+
+    assertThat(output, containsString("Dependency updates available:"));
+    assertThat(output, containsString("org#example1  1.0 -> 2.0"));
+    assertThat(output, not(containsString("org#example2 (transitive)  2.0 -> 2.1")));
+  }
+
+  @Test
+  public void testTransitiveDependencies() throws Exception {
+    testArgs.add(IVY_FILE_FLAG);
+    testArgs.add("tests/resources/org/pantsbuild/tools/ivy/ivy.xml");
+    testArgs.add(SHOW_TRANSITIVE_FLAG);
+
+    String output = runDependencyUpdateChecker(testArgs);
+
+    assertThat(output, containsString("Dependency updates available:"));
+    assertThat(output, containsString("org#example1  1.0 -> 2.0"));
+    assertThat(output, containsString("org#example2 (transitive)  2.0 -> 2.1"));
+  }
+
+  @Test
+  public void testDependencyThatIsMissingIvyFile() throws Exception {
+    testArgs.add(IVY_FILE_FLAG);
+    testArgs.add("tests/resources/org/pantsbuild/tools/ivy/ivy-dep-missing-ivy-file.xml");
+
+    String output = runDependencyUpdateChecker(testArgs);
+
+    assertThat(output, containsString("Dependency updates available:"));
+    assertThat(output, containsString("org#example2  2.0 -> 2.1"));
+  }
+
+  @Test
+  public void testExcludedConf() throws Exception {
+    testArgs.add(IVY_FILE_FLAG);
+    testArgs.add("tests/resources/org/pantsbuild/tools/ivy/ivy.xml");
+    testArgs.add(CONFS_FLAG);
+    testArgs.add("*,!default");
+
+    String output = runDependencyUpdateChecker(testArgs);
+
+    assertThat(output, containsString("Dependency updates available:"));
+    assertThat(output, containsString("All dependencies are up to date"));
+  }
+
+  @Test
+  public void testUnknownDependency() throws Exception {
+    testArgs.add(IVY_FILE_FLAG);
+    testArgs.add("tests/resources/org/pantsbuild/tools/ivy/ivy-unknown-dep.xml");
+
+    try {
+      runDependencyUpdateChecker(testArgs);
+      fail("Expected RuntimeException");
+    } catch (RuntimeException e) {
+      assertThat(e.toString(), containsString("DependencyUpdateChecker exited with status"));
+    }
+  }
+
+  @Test
+  public void testUnknownConfigurations() throws Exception {
+    testArgs.add(IVY_FILE_FLAG);
+    testArgs.add("tests/resources/org/pantsbuild/tools/ivy/ivy.xml");
+    testArgs.add(CONFS_FLAG);
+    testArgs.add("default,unknown");
+
+    try {
+      runDependencyUpdateChecker(testArgs);
+      fail("Expected IllegalArgumentException");
+    } catch (IllegalArgumentException e) {
+      assertThat(e.toString(), containsString("requested configuration not found"));
+    }
+  }
+
+  @Test
+  public void testDependencyWithInvalidIvyFile() throws Exception {
+    testArgs.add(IVY_FILE_FLAG);
+    testArgs.add("tests/resources/org/pantsbuild/tools/ivy/ivy-dep-with-invalid-ivy.xml");
+
+    try {
+      runDependencyUpdateChecker(testArgs);
+      fail("Expected RuntimeException");
+    } catch (RuntimeException e) {
+      assertThat(e.toString(), containsString("DependencyUpdateChecker exited with status"));
```

Comment:
The only thing you get is the RuntimeException without much information.  The rest of the error is in the output which isn't captured in the test because the LogStream doesn't get converted to a string when an exception is raised.